lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php

```diff
@@ -444,10 +444,10 @@ private function getTableDiffForAlterForeignKey(ForeignKeyConstraint $foreignKey
     private function parseColumnCollationFromSQL($column, $sql)
     {
         if (preg_match(
-            '{(?:'.preg_quote($column).'|'.preg_quote($this->_platform->quoteSingleIdentifier($column)).')
+            '/(?:' . preg_quote($column, '#') . '|' . preg_quote($this->_platform->quoteSingleIdentifier($column), '#') . ')
```

Comment:
I still strongly disagree with abusing `preg_quote()` to escape single character. It should be done by `str_replace()` or `addcslashes()`.
Also, changing delimiter to `/` introduces another issue - you have to escape `/` as well, which you can't do if you abuse `preg_quote()`'s 2nd parameter to escape `#`.
So as it currently stands, this would be correct:
```php
addcslashes(preg_quote($column, '/'), '#')
```
https://3v4l.org/8Bnvj